### PR TITLE
Feature/test

### PR DESCRIPTION
**Pull request by Gnwin**
## Math Magicians: tests

In this milestone, the full website of the application was tested using jest and react testing library by me and my partner.  All requirements for this milestone have been met according to this document:

https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/project_tests.md

### CODE REVIEW

File: src/reactjs/__tests__/App.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    const appComponent = screen.getByTestId('app-1');
+    expect(appComponent).toBeInTheDocument();
+  });
+
+  it('renders the home page in the App component when the user clicks the home link', () => {
+    render(<BrowserRouter><App /></BrowserRouter>);
+    const homeLink = screen.getByText('Home');
+    fireEvent.click(homeLink);
+    const homeContainer = screen.getByTestId('home-1');
+    const firstchild = homeContainer.childNodes[0];
```
Comment (Amrhub) [inline src/reactjs/__tests__/App.test.js:27]: - [ ] [OPTIONAL] To follow best practices for JavaScript let's kindly make sure we name variable with camel case and be consistent with it I could see you did a great job at naming `homeContainer` but when it came to `firstchild` wasn't camel case this is not the only example in the project but having this in mind that would help you later on to follow best practices easily I hope ☺️

File: src/reactjs/__tests__/CalculatorPage.test.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  it('checks if calcpageTitle title is a descendant of calcpageComponent', () => {
+    render(<CalculatorPage />);
+    const calcpageComponent = screen.getByTestId('calc-1');
+    const calcpageTitle = screen.getByTestId('calc-2');
+    expect(calcpageComponent).toContainElement(calcpageTitle);
+  });
+
+  it('shows 8 in the answer cell when the user clicks the 8 button', () => {
+    render(<BrowserRouter><App /></BrowserRouter>);
```
Comment (Amrhub) [inline src/reactjs/__tests__/CalculatorPage.test.js:25]: - [ ] [OPTIONAL] I could see we need this line to be present in all next tests so I would highly suggest using setup and teardown in jest, which will save us time and code if we extended the testing we don't have to repeat this code :shipit: 
```js
beforeEach( () => {
render(<BrowserRouter><App /></BrowserRouter>);
});
```
[Setup and Teardown](https://jestjs.io/docs/setup-teardown)

**Review by Amrhub — APPROVED** [review]
# Approved 💯 ✅ 🥳 🎈 🎉 

Hi Team!, 👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights ✅ 
Great job covering all tests ✔️ 
Highly organized and clean code ✔️ 👨🏻‍💻 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Gnwin** [comment]
Thanks @Amrhub

**Pull request merged**